src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -210,6 +212,14 @@ class ReportActionItemMessageEdit extends React.Component {
                             ReportScrollManager.scrollToIndex({animated: true, index: this.props.index}, true);
                             toggleReportActionComposeView(false, VirtualKeyboard.shouldAssumeIsOpen());
                         }}
+                        onBlur={(event) => {
+                            // Return to prevent re-render when save/cancel button is pressed which cancels the onPress event by re-rendering
+                            if (_.contains([this.saveButtonID, this.cancelButtonID], lodashGet(event, 'nativeEvent.relatedTarget.id'))) {
+                                return;
+                            }
+
+                            toggleReportActionComposeView(true, VirtualKeyboard.shouldAssumeIsOpen());
+                        }}
```

Comment:
@marcaaron Could you please help me understand this change? It has caused an issue and I am trying to figure out what is it doing? Especially What do I need to test for this?

Reply:
What are you curious about exactly?

Reply:
Want to maybe walk me through what you think it's doing?

Reply:
So it is checking that if the Cancel or save button is clicked on the Editor, return, otherwise show the composer. 

So do we want to show the composer on blur?

Reply:
IIRC, in this case we are ignoring the blur event because when we are blurring because the cancel or save button is clicked the method to toggle the composer (`toggleReportActionComposeView`) would prevent the user from taking the actions. I assumed here (perhaps incorrectly) that it is related to re-rendering of the component - but the `onPress` method for the buttons never seemed to run. Let me know if that helps!

Reply:
Got, It. Thanks.

Reply:
Do we want to show the composer on blur?

Reply:
Yes on blur and it should happen when we save changes or cancel the edit right? Can we have this conversation on whatever issue you are working on so I can get more insight into what you're trying to figure out?

Reply:
@marcaaron i got your [point](https://github.com/Expensify/App/pull/9060#:~:text=in%20this%20case%20we%20are%20ignoring%20the%20blur%20event%20because%20when%20we%20are%20blurring%20because%20the%20cancel%20or%20save%20button%20is%20clicked%20the%20method%20to%20toggle%20the%20composer%20(toggleReportActionComposeView)%20would%20prevent%20the%20user%20from%20taking%20the%20actions). Actually onBlur() is invoked when `emojiPicker` button is pressed and it shouldn't and on blur of `emojiPicker` we should only toggle `ReportActionComposeViewEdit`. 

- [This](https://github.com/Expensify/App/issues/9252#:~:text=Return%20from%20onBlur%20to%20prevent%20re%2Drender%20when%20emoji%20button%20is%20pressed%20which%20cancels%20the%20onPress%20event%20by%20re%2Drendering.%20if%20(_.contains(%5Bthis.saveButtonID%2C%20this.cancelButtonID) is the issue to give you more insight

Reply:
Please bring whatever conversation you'd like to have to a relevant issue.